src/service.rs

```diff
@@ -393,9 +403,77 @@ where
     Ok(Body::from_json(&f().await?)?.into())
 }
 
+async fn do_reset(mnemonic: SecretString, password: SecretString, config: &Config) -> State {
+    info!("resetting signer state").expect("Logger error in reset handler");
+
+    let mnemonic = Mnemonic::new(mnemonic.expose_secret()).expect("Invalid mnemonic phrase");
+    let password_hash = PasswordHash::<Argon2>::from_default(password.expose_secret().as_bytes());
+
+    let data_path = config.data_directory().to_owned();
+    let parameters = task::spawn_blocking(move || crate::parameters::load(data_path))
+        .await
+        .expect("Parameter loading error")
+        .expect("Failed to load parameters");
+
+    let state = SignerState::new(
+        TestnetKeySecret::new(mnemonic, password.expose_secret())
+            .map(HierarchicalKeyDerivationFunction::default()),
+        UtxoAccumulator::new(
+            task::spawn_blocking(crate::parameters::load_utxo_accumulator_model)
+                .await
+                .expect("Error loading utxo accumulator model")
+                .unwrap(),
+        ),
+    );
+    info!("saving signer state").expect("Logger error in reset handler");
+    let password_hash_bytes = password_hash.as_bytes();
+    let cloned_state = state.clone();
+    let path = config.data_path.clone();
+
+    task::spawn_blocking(move || File::save(&path, &password_hash_bytes, cloned_state))
+        .await
+        .expect("Signer state save failed")
+        .expect("Could not save signer state.");
```

Comment:
Much of this code is repeated in other parts of `service.rs`. Please find a refactoring that does not introduce repetition (even if that means writing new functions that generalize over all the use cases).